frontend/class-opengraph.php

```diff
@@ -386,103 +381,17 @@ public function type( $echo = true ) {
 	}
 
 	/**
-	 * Display an OpenGraph image tag
-	 *
-	 * @param string $img Source URL to the image
-	 *
-	 * @return bool
-	 */
-	public function image_output( $img ) {
-		/**
-		 * Filter: 'wpseo_opengraph_image' - Allow changing the OpenGraph image
-		 *
-		 * @api string $img Image URL string
-		 */
-		$img = trim( apply_filters( 'wpseo_opengraph_image', $img ) );
-
-		if ( empty( $img ) ) {
-			return false;
-		}
-
-		if ( WPSEO_Utils::is_url_relative( $img ) === true ) {
-			if ( $img[0] != '/' ) {
-				return false;
-			}
-
-			// If it's a relative URL, it's relative to the domain, not necessarily to the WordPress install, we
-			// want to preserve domain name and URL scheme (http / https) though.
-			$parsed_url = parse_url( home_url() );
-			$img        = $parsed_url['scheme'] . '://' . $parsed_url['host'] . $img;
-		}
-
-		if ( in_array( $img, $this->shown_images ) ) {
-			return false;
-		}
-
-		array_push( $this->shown_images, $img );
-
-		$this->og_tag( 'og:image', esc_url( $img ) );
-
-		return true;
-	}
-
-	/**
-	 * Output the OpenGraph image elements for all the images within the current post/page.
-	 *
-	 * @return bool
+	 * Create new WPSEO_OpenGraph_Image class and get the images to set the og:image
 	 */
 	public function image() {
+		$opengraph_images = new WPSEO_OpenGraph_Image( $this->options );
 
-		global $post;
-
-		if ( is_front_page() ) {
-			if ( $this->options['og_frontpage_image'] !== '' ) {
-				$this->image_output( $this->options['og_frontpage_image'] );
-			}
+		foreach ( $opengraph_images->get_images() as $img ) {
```

Comment:
the function get_images always returns an array. It'll return an empty array if there are no images.